### PR TITLE
[server] Do not set data recovery for DaVinci client

### PR DESCRIPTION
**Pull request by majisourav99**
##  Do not set data recovery for DaVinci client

During target colo push for stores with DaVinci client, or during data recovery,  Venice uses data recovery client endpoints to enable push to rest of the colos, but for DaVinci client running in follower mode does not have the configs needed for data recovery which leads to DaVinci host ingestion failure. 
This PR skip data recovery code path in Davinci clients.

## How was this PR tested?

GHCI
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by gaojieliu — COMMENTED** [review]
Can we have an E2E test for this logic?

**Comment by majisourav99** [comment]
> Can we have an E2E test for this logic?

Sure, dont think we tested data recovery code path with DVC client. That should also could catch it.

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java [hunk 1]
```diff
@@ -2,14 +2,7 @@
...
 import static com.linkedin.venice.CommonConfigKeys.SSL_ENABLED;
-import static com.linkedin.venice.ConfigKeys.DEFAULT_MAX_NUMBER_OF_PARTITIONS;
-import static com.linkedin.venice.ConfigKeys.EMERGENCY_SOURCE_REGION;
-import static com.linkedin.venice.ConfigKeys.PUSH_JOB_STATUS_STORE_CLUSTER_NAME;
-import static com.linkedin.venice.ConfigKeys.SERVER_DATABASE_CHECKSUM_VERIFICATION_ENABLED;
-import static com.linkedin.venice.ConfigKeys.SERVER_DATABASE_SYNC_BYTES_INTERNAL_FOR_DEFERRED_WRITE_MODE;
-import static com.linkedin.venice.ConfigKeys.SERVER_KAFKA_PRODUCER_POOL_SIZE_PER_KAFKA_CLUSTER;
-import static com.linkedin.venice.ConfigKeys.SERVER_PROMOTION_TO_LEADER_REPLICA_DELAY_SECONDS;
-import static com.linkedin.venice.ConfigKeys.SERVER_SHARED_KAFKA_PRODUCER_ENABLED;
+import static com.linkedin.venice.ConfigKeys.*;
```
Comment (gaojieliu) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java:5]: Let us not use wildcard import?

**Review by gaojieliu — CHANGES_REQUESTED** [review]

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java (hunk 1 above)
Comment (majisourav99) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java:5]: Its not annoying, intellij always forget this setting after every restart.

**Review by gaojieliu — APPROVED** [review]

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java [hunk 2]
```diff
@@ -993,6 +1003,22 @@ public void testTargetedRegionPushJobFullConsumptionForBatchStore() throws Excep
...
             });
           }
+          startDaVinciClient(storeName);
+          props.put(TARGETED_REGION_PUSH_ENABLED, true);
+          props.put(POST_VALIDATION_CONSUMPTION_ENABLED, true);
+          try (VenicePushJob job = new VenicePushJob("Test push job 2", props)) {
+            job.run(); // the job should succeed
+
+            TestUtils.waitForNonDeterministicAssertion(45, TimeUnit.SECONDS, () -> {
+              // Current version should become 1
```
[the remark this answers is not in the document]
Comment (majisourav99) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java]: added dvc client validation.

**Pull request merged**